src/Features/Core/Portable/DesignerAttribute/AbstractDesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -143,47 +152,29 @@ public DesignerAttributeDiscoveryService()
 
                 // If nothing has changed at the top level between the last time we analyzed this document and now, then
                 // no need to analyze again.
+                var projectVersion = await lazyProjectVersion.GetValueAsync(cancellationToken).ConfigureAwait(false);
                 if (_documentToLastReportedInformation.TryGetValue(document.Id, out var existingInfo) &&
                     existingInfo.projectVersion == projectVersion)
                 {
                     continue;
                 }
 
-                tasks.Add(ComputeDesignerAttributeDataAsync(designerCategoryType, document, cancellationToken));
+                var data = await ComputeDesignerAttributeDataAsync(document).ConfigureAwait(false);
+                if (data.Category != existingInfo.category)
+                    results.Add((data, projectVersion));
             }
 
-            using var _2 = ArrayBuilder<DesignerAttributeData>.GetInstance(tasks.Count, out var results);
```

Comment:
removed parallel processing here.  this runs in the bg, and uses resources that active features (like lightbulbs) want to be using.